### PR TITLE
Refactor to simplify strategy creation

### PR DESCRIPTION
**Pull request by gndelia**
This refactor simplifies strategy creation, by having the instantiation of the caches taking place in one file. It also extends the code so strategies could resolve to other caches created by other strategies. 

`index.ts` was renamed to `rpc.ts` so the index file could export the strategy files

### CODE REVIEW

File: packages/eth-rpc-cache/src/strategies/per-block.ts [hunk 1]
```diff
@@ -35,20 +30,8 @@ const perBlockMethods = [
...
-  pMemoize(rpc, {
-    cache,
-    maxAge: 6000, // Half block time: ~6 sec.
-    resolver: (method: string, params: unknown[]) => getKey(method, params),
-    ...options
-  })
-
 export const perBlockStrategy: Strategy = {
-  getRpc,
+  maxAge: 6000, // Half block time: ~6 sec.
```
Comment (gndelia) [inline packages/eth-rpc-cache/src/strategies/per-block.ts:34]: I wonder if we should convert this into a constructor function, so `maxAge` can be parameterized

```js
const perBlockStrategy = ({ maxAge = 6000 }) => ({
  maxAge,
  methods,
  name: 'block'
})
```

This way it is exensible for chains that use a different block time and are evm compatible (For `permanent` strategy, there would be no parameters, but I would do it as a constructor still for consistency)

File: packages/eth-rpc-cache/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eth-rpc-cache",
-  "version": "1.0.0",
+  "version": "1.1.0",
```
Comment (gabmontes) [inline packages/eth-rpc-cache/package.json:3]: If someone is using v1.0.0 with a custom strategy, these changes will most probably break their code. We should bump it to 2.0.0 to be safe.

File: packages/eth-rpc-cache/src/strategies/per-block.ts (hunk 1 above)
Comment (gabmontes) [reply packages/eth-rpc-cache/src/strategies/per-block.ts:34]: This override can be done today through the `options` passed to `createEthRpcCache()`. Let's keep thinking on how to better address this problem for a future version.

Comment (gndelia) [reply packages/eth-rpc-cache/src/strategies/per-block.ts:34]: true! We actually discussed about it, I forgot it 😆

File: packages/eth-rpc-cache/package.json (hunk 2 above)
Comment (gndelia) [reply packages/eth-rpc-cache/package.json]: version updated [fc17775](https://github.com/bloq/eth-packages/pull/6/commits/fc1777506dcbdb09a6c043fbbe886b5946f267f0)

**Review by gabmontes — APPROVED** [review]

**Pull request merged**